pulp_rpm/app/shared_utils.py

```diff
@@ -25,6 +25,8 @@ def _prepare_package(artifact, filename):
         filename: name of file uploaded by user
     """
     artifact_file = storage.open(artifact.file.name)
+    # replace all slashes in filename as it can result in wrong path
+    filename = filename.replace("/", "_")
```

Comment:
This seems like a completely different change?  And it would prevent us from fixing up the `location_href` to be just a filename by using `os.path.basename()`

Reply:
If you can just use `os.path.basename()` here, that would be ideal.